app/models/message_object.rb

```diff
@@ -135,4 +136,12 @@ def unassign_tag(tag)
   def thread
     message.thread
   end
+
+  def remove_object_related_tags_from_thread
+    tags.each do |tag|
+      message.thread.unassign_tag(tag) unless message.thread.objects.where.not(id: id).map(&:tags).flatten&.map(&:id).include?(tag.id)
+    end
+
+    message.thread.unassign_tag(message.tenant.signed_tag!) unless message.thread.tags.reload.where(type: SignedByTag.to_s).any?
```

Comment:
Ak toto pouzivame castejsie tak mozno dava zmysel spravit z toho scope `with_signed_by_tag`

Reply:
Mozeme spravit, ale pozerala som, zda sa, ze to castejsie nepouzivame.